test/jdk/tools/jpackage/share/jdk/jpackage/tests/NoMPathRuntimeTest.java

```diff
@@ -0,0 +1,160 @@
+
+
+/*
+ * @test
+ * @summary test '--runtime-image' option of jpackage
+ * @library ../../../../helpers
+ * @build jdk.jpackage.test.*
+ * @modules jdk.jpackage/jdk.jpackage.internal
+ * @compile NoMPathRuntimeTest.java
+ * @run main/othervm/timeout=360 -Xmx512m jdk.jpackage.test.Main
+ *  --jpt-run=jdk.jpackage.tests.NoMPathRuntimeTest
+ */
+
+public final class NoMPathRuntimeTest {
+
+    public NoMPathRuntimeTest(String javaAppDesc, String jlinkOutputSubdir,
+            String runtimeSubdir) {
+        this.javaAppDesc = javaAppDesc;
+        this.jlinkOutputSubdir = Path.of(jlinkOutputSubdir);
+        this.runtimeSubdir = Path.of(runtimeSubdir);
+    }
+
+    @Test
+    public void test() throws IOException {
+        final Path tmpdir = TKit.createTempDirectory("tmpdir");
+        try {
+            Files.createFile(tmpdir.resolve("tmpfile"));
+        } catch (IOException ioe) {
+            // ignore
+        }
+        JavaAppDesc appDesc = JavaAppDesc.parse(javaAppDesc);
+
+        JPackageCommand cmd = JPackageCommand.helloAppImage(appDesc);
+
+        final String moduleName = appDesc.moduleName();
+
+        if (moduleName != null) {
+            // Build module jar.
+            cmd.executePrerequisiteActions();
+        }
+
+        final Path workDir = TKit.createTempDirectory("runtime").resolve("data");
+        final Path jlinkOutputDir = workDir.resolve(jlinkOutputSubdir);
+        Files.createDirectories(jlinkOutputDir.getParent());
+
+        // List of modules required for test app.
+        final var modules = new String[] {
+            "java.base",
+            "java.desktop"
+        };
+
+        Executor jlink = new Executor()
+        .setToolProvider(JavaTool.JLINK)
+        .dumpOutput()
+        .addArguments(
+                "--add-modules", String.join(",", modules),
+                "--output", jlinkOutputDir.toString(),
+                "--strip-debug",
+                "--no-header-files",
+                "--no-man-pages");
+
+        if (moduleName != null) {
+            jlink.addArguments("--add-modules", moduleName, "--module-path",
+                    Path.of(cmd.getArgumentValue("--module-path")).resolve(
+                            "hello.jar").toString());
+        }
+
+        jlink.execute();
+
+        ArrayList<Path> paths = new ArrayList<Path>();
+        paths.add(TKit.TEST_SRC_ROOT.resolve("apps/image/Hello.java"));
+
+        // compile Hello.java again putting classes in tmpdir
+        new Executor()
+                .setToolProvider(JavaTool.JAVAC)
+                .addArguments("-d", tmpdir.toString())
+                .addPathArguments(paths)
+                .execute();
+
+        // make raw jar of just Hello.class from tmpdir.
+        new Executor()
+                .setToolProvider(JavaTool.JAR)
+                .dumpOutput()
+                .addArguments("-cvf", "junk.jar",
+                        "-C", tmpdir.toString(), "Hello.class")
+                .execute();
```

Comment:
Single line `HelloApp.createBundle("junk.jar:Hello", tmpdir);` would compile source class and put it into "junk.jar" in `tmpdir` folder. It can be used to replace lines from [109, 125] range.

Reply:
The bug is that when --module-path option is not used in a modular app, jpackage uses a module-path with "." on it.
Having a non-modular jar in the modular path is an error.
So with this non-modular Hello.jar in the current directory the jpackage command failed before the fix, and succeeds after the fix.

I can create the non-modular Hello.jar in the current directory with one line:
HelloApp.createBundle(JavaAppDesc.parse("junk.jar:Hello"), Path.of("."))

Reply:
Another precondition for the test is that Java runtime used with jpackage command should include module with app main class, right?
Test arguments are: `List.of("Hello", "com.foo/com.foo.main.Aloha");`. The first argument is non-modular app, it is not used with jlink. What is the point to run the test for non-modular app?